### PR TITLE
On create JS

### PR DESCRIPTION
**Pull request by SHeimlich**
Add the Particle onCreate method to Javascript.

### CODE REVIEW

File: src/runtime/tests/particle-interface-loading-test.ts [hunk 1]
```diff
@@ -235,4 +237,54 @@ describe('particle interface loading', () => {
...
+    await arc.idle;
+    const fooHandle = await singletonHandleForTest(arc, fooStore);
+    assert.deepStrictEqual(await fooHandle.get(), {value: 'Created!'});
+
+    const serialization = await arc.serialize();
+
+    const arc2 = await Arc.deserialize({serialization, loader, fileName: '', context: manifest});
+    await arc2.idle;
+
+    const fooHandle2 = await singletonHandleForTest(arc2, arc2._stores[1]);
```
Comment (piotrswigon) [inline src/runtime/tests/particle-interface-loading-test.ts]: `arc2._stores[1]` looks a bit magical!

Comment (piotrswigon) [reply src/runtime/tests/particle-interface-loading-test.ts]: Any way to improve it?

File: src/runtime/tests/particle-interface-loading-test.ts [hunk 2]
```diff
@@ -235,4 +237,54 @@ describe('particle interface loading', () => {
     const fooHandle = await singletonHandleForTest(arc, fooStore);
     assert.deepStrictEqual(await fooHandle.get(), {value: 'hello world!!!'});
   });
+
+  it('test onCreate only runs for initialization and not reinstantiation', async () => {
```
Comment (csilvestrini) [inline src/runtime/tests/particle-interface-loading-test.ts]: Nit: delete `test`

File: src/runtime/tests/particle-interface-loading-test.ts (hunk 1 above)
Comment (SHeimlich) [reply src/runtime/tests/particle-interface-loading-test.ts]: I think this is better... thoughts?

File: src/runtime/tests/particle-interface-loading-test.ts [hunk 3]
```diff
@@ -235,4 +235,62 @@ describe('particle interface loading', () => {
...
+    await arc.idle;
+    const fooHandle = await singletonHandleForTest(arc, fooStore);
+    assert.deepStrictEqual(await fooHandle.fetch(), {value: 'Created!'});
+
+    const serialization = await arc.serialize();
+
+    const arc2 = await Arc.deserialize({serialization, loader, fileName: '', context: manifest});
+    await arc2.idle;
+
+    let fooStore2 = fooStore;
```
Comment (piotrswigon) [inline src/runtime/tests/particle-interface-loading-test.ts]: maybe:
```
fooStore2 = arc2._stores.find(s => s.type.tag === 'Entity');
```

File: src/runtime/tests/particle-interface-loading-test.ts (hunk 1 above)
Comment (piotrswigon) [reply src/runtime/tests/particle-interface-loading-test.ts]: Thank you :)

**Comment by SHeimlich** [comment]
Includes workaround to issue #4712

**Pull request merged**